### PR TITLE
MacOS support

### PR DESCRIPTION
**Pull request by dartasen**

### CODE REVIEW

File: Nitrox.Launcher/Program.cs [hunk 1]
```diff
@@ -51,11 +52,24 @@ private static AppBuilder BuildAvaloniaApp()
...
     {
-        using ProcessEx process = ProcessEx.GetFirstProcess("Nitrox.Launcher", process => process.Id != Environment.ProcessId);
-        if (process != null)
+        // This feature is windows only
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
         {
-            process.SetForegroundWindowAndRestore();
-            Environment.Exit(0);
+            return;
+        }
```
Comment (Measurity) [inline Nitrox.Launcher/Program.cs:59]: Code on the branch now has linux support here too. But not OSX yet.

File: Nitrox.Launcher/Platforms/MacOS/Info.plist [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        <key>CFBundlePackageType</key>
+        <string>APPL</string>
+        <key>CFBundleShortVersionString</key>
+        <string>1.8</string>
+        <key>NSPrincipalClass</key>
+        <string>NSApplication</string>
+        <key>NSHighResolutionCapable</key>
+        <true/>
+    </dict>
+</plist>
```
Comment (Measurity) [inline Nitrox.Launcher/Platforms/MacOS/Info.plist:30]: Would be good to generate this from within msbuild (and embed it at compile target?). Then the info can always sync with msbuild properties.

(Though I'm unfamiliar with how plist is used by OSX and why it's in `/Platforms/MacOS/`.)

Comment (dartasen) [reply Nitrox.Launcher/Platforms/MacOS/Info.plist:30]: Good idea, could definitely be generated by hand, that's what Xamarin, MAUI, ... are doing under the hood.

Info.Plist it's just a manifest in apple ecosystem that will tell information about the application, Name, Version, Mininum required OS version, Executable Name, Executable type.
Entitlement will just ask for specific permissions (android like), those are mainly used when apple need to notarize one of your app during review session

Comment (dartasen) [reply Nitrox.Launcher/Platforms/MacOS/Info.plist:30]: Will reconsider adding manifest generation once this once is done

**Review by Measurity — APPROVED** [review]
LGTM :dancers:

**Pull request merged**